### PR TITLE
[fuchsia] Expose view_ref as part of dart:fuchsia initialization

### PR DESCRIPTION
**Pull request by iskakaushik**
fixes https://github.com/flutter/flutter/issues/47178

### CODE REVIEW

**Comment by iskakaushik** [comment]
@chinmaygarde , @arbreng I will add tests for these, wanted to give you a heads-up on the approach i'm taking.

File: shell/platform/fuchsia/flutter/isolate_configurator.h [hunk 1]
```diff
@@ -21,7 +21,8 @@ class IsolateConfigurator final {
   IsolateConfigurator(
       UniqueFDIONS fdio_ns,
       fidl::InterfaceHandle<fuchsia::sys::Environment> environment,
-      zx::channel directory_request);
+      zx::channel directory_request,
+      zx::eventpair view_ref);
```
[the remark this answers is not in the document]
Comment (iskakaushik) [reply shell/platform/fuchsia/flutter/isolate_configurator.h]: Good call, I will make the optional change.

File: shell/platform/fuchsia/dart-pkg/fuchsia/lib/fuchsia.dart [hunk 2]
```diff
@@ -40,6 +43,16 @@ class MxStartupInfo {
     _outgoingServices = null;
     return handle;
   }
+
+  static Handle takeViewRef() {
```
[the remark this answers is not in the document]
Comment (iskakaushik) [reply shell/platform/fuchsia/dart-pkg/fuchsia/lib/fuchsia.dart]: I don't think there is, I will create one.

**Comment by gw280** [comment]
LGTM modulo David's comments

**Comment by iskakaushik** [comment]
Landing this with the intent of adding tests once we have the test framework ready. Filed: https://github.com/flutter/flutter/issues/49447

**Pull request merged**